clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -337,14 +337,26 @@ protected RegisterTaskDefinitionRequest makeTaskDefinitionRequest(
   protected RegisterTaskDefinitionRequest makeTaskDefinitionRequestFromArtifact(
       String ecsServiceRole, EcsServerGroupName newServerGroupName) {
 
-    File artifactFile =
-        downloadTaskDefinitionArtifact(description.getResolvedTaskDefinitionArtifact());
-
-    RegisterTaskDefinitionRequest requestTemplate;
-    try {
-      requestTemplate = mapper.readValue(artifactFile, RegisterTaskDefinitionRequest.class);
-    } catch (IOException e) {
-      throw new UncheckedIOException(e);
+    RegisterTaskDefinitionRequest requestTemplate = null;
+    if (!description.isEvaluateTaskDefinitionArtifactExpressions()) {
```

Comment:
this would be more readable if we evaluated the `true` condition first and extracted the logic of each block to separate functions. then we could end up with something like:
```java
RegisterTaskDefinitionRequest requestTemplate = null;
if (description.isEvaluateTaskDefinitionArtifactExpressions()) {
    requestTemplate = getSpelProcessedArtifact();
} else {
    requestTemplate = getArtifactFromFile();
}
```

Reply:
done.